addons/MountRoulette/MountRoulette.lua

```diff
@@ -0,0 +1,79 @@
+_addon.name    = 'Mount Roulette'
+_addon.author  = 'Dean James (Xurion of Bismarck)'
+_addon.version = '3.0.0'
+_addon.commands = {'mountroulette', 'mr'}
+
+require('lists')
+resources = require('resources')
+
+math.randomseed(os.time())
+
+allowed_mounts = L{}
+possible_mounts = L{}
+for _, mount in ipairs(resources.mounts) do
+    possible_mounts:append(mount.name:lower())
+end
+
+function update_allowed_mounts()
+    local kis = windower.ffxi.get_key_items()
+
+    for _, id in ipairs(kis) do
+        local ki = resources.key_items[id]
+        if ki.category == 'Mounts' and ki.name ~= "trainer's whistle" then -- Don't care about the quest KI
+            local mount_index = possible_mounts:find(function(possible_mount)
+                return windower.wc_match(ki.name:lower(), '♪' .. possible_mount .. '*')
+            end)
+            local mount = possible_mounts[mount_index]
+
+            -- Add this to allowed mounts if it is not already there
+            if not allowed_mounts:contains(mount) then
+                allowed_mounts:append(mount)
+            end
+        end
+    end
+end
```

Comment:
I think this function needs to clear the `allowed_mounts` every time it's called. Otherwise if you log into a character that has all mounts, then log out and into a mule that has none, it will still have all mounts unlocked, no? Or am I missing something here?

Reply:
You're absolutely right, I'll clear it.